smithy-model/src/main/java/software/amazon/smithy/model/FromSourceLocation.java

```diff
@@ -28,4 +28,15 @@ public interface FromSourceLocation {
     default SourceLocation getSourceLocation() {
         return SourceLocation.none();
     }
+
+    /**
+     * Compares two SourceLocations.
+     *
+     * @param s1 the first SourceLocation to compare.
+     * @param s2 the second SourceLocation to compare.
+     * @return the value 0 if s1 == s2; a value less than 0 if s1 < s2; and a value greater than 0 if s1 > s2.
+     */
+    static int compare(SourceLocation s1, SourceLocation s2) {
```

Comment:
This should take two FromSourceLocations and compare them by getting the source location of s1 and calling compareTo with the sourcelocaiton of s2

Reply:
Fixed in  bb54034